pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -50,6 +50,16 @@ type KubernetesResources struct {
 	Domain string
 }
 
+// gatewayLabelSelectorAsSelector is like metav1.LabelSelectorAsSelector but for gateway selector, which
+// has different logic
+func gatewayLabelSelectorAsSelector(ps *metav1.LabelSelector) (klabels.Selector, error) {
```

Comment:
Are there plans to add this to k8s?

Reply:
In k8s - probably never, since its gateway-specific.
In gateway API - I think there is a lot of common helpers that should probably live in their repo but currently its strictly the API, so there isn't an obvious place right now. 

I think the plan in v1alpha2 is to remove this anyways so long term it doesn't matter too much

Reply:
The entire larger function this is used could be in upstream ideally, for example